lib/hotel_controller.rb

```diff
@@ -0,0 +1,65 @@
+require_relative 'date_range'
+require_relative 'reservation'
+require_relative 'room'
+
+module Hotel
+  class Hotel::Controller
+    attr_reader :rooms
+
+    def initialize
+      create_rooms
+    end
+
+    def find_available_rooms(input_range)
+      available = []
+      @rooms.each do |room|
+        if room.conflict?(input_range) == false
+          available << room.room_id
+        end
+      end
+
+      return available
+    end
+
+    def reserve_with_range(input_range)
+      @rooms.each do |room|
+        return room.room_id if room.create_room_reservation(input_range)
+      end
+
+      raise ArgumentError.new("No rooms available for date range.")
```

Comment:
I'd argue that this isn't _really_ an `ArgumentError`. The argument's fine, but your hotel is full. This is a great place for a custom exception!